src/sentry/static/sentry/app/components/group/releaseStats.tsx

```diff
@@ -1,28 +1,28 @@
-import PropTypes from 'prop-types';
 import React from 'react';
 
-import SentryTypes from 'app/sentryTypes';
 import LoadingIndicator from 'app/components/loadingIndicator';
 import GroupReleaseChart from 'app/components/group/releaseChart';
 import SeenInfo from 'app/components/group/seenInfo';
 import getDynamicText from 'app/utils/getDynamicText';
 import {t} from 'app/locale';
+import {Environment, Group, Organization, Project} from 'app/types';
 
-class GroupReleaseStats extends React.PureComponent {
-  static propTypes = {
-    group: SentryTypes.Group.isRequired,
-    project: SentryTypes.Project.isRequired,
-    organization: SentryTypes.Organization.isRequired,
-    environments: PropTypes.arrayOf(SentryTypes.Environment).isRequired,
-    allEnvironments: PropTypes.object,
-  };
+type Props = {
+  organization: Organization;
+  project: Project;
+  environments: Environment[];
+  allEnvironments: Group | undefined;
+  group: Group | undefined;
+};
 
+class GroupReleaseStats extends React.PureComponent<Props> {
```

Comment:
As you are not using a component's state and any life cycle method, it might be a could idea to convert it to functional component and use React.memo 

> React.memo is a higher order component. It’s similar to React.PureComponent but for function components instead of classes.

Reply:
Ooo nice, I'll look deeper into this in the morning 🙏

Reply:
Added React.memo to try it out. Is this something that we want to use more widely across the codebase?

Reply:
Oh, I saw that you brought this topic to the TSC meeting yesterday .. unfortunately I did not participate, but I agree that you should use React.memo or React.useMemo (hook) only if you need to reduce frequent re-renders 🙂